modules/gateway/rpc.go

```diff
@@ -155,3 +154,50 @@ func (g *Gateway) Broadcast(name string, obj interface{}) {
 	}
 	wg.Wait()
 }
+
+// Broadcast calls an RPC on all of the peers in the Gateway's peer list. The
+// calls are run in parallel. Broadcasts are restricted to "one-way" RPCs,
+// which simply write an object and disconnect. This is why Broadcast takes an
+// interface{} instead of an RPCFunc.
+func (g *Gateway) Broadcast(name string, obj interface{}) {
+	peers := g.Peers()
+	g.broadcastTo(name, obj, peers)
+}
+
+// BroadcastMinVersion calls an RPC on all of the peers in the Gateway's peer
+// list that are at least the specified version. The calls are run in parallel.
+// Broadcasts are restricted to "one-way" RPCs, which simply write an object
+// and disconnect. This is why Broadcast takes an interface{} instead of an
+// RPCFunc. BroadcastMinVersion can be used to ensure new RPC calls are not
+// made to old nodes that wouldn't understand them.
+func (g *Gateway) BroadcastMinVersion(name string, obj interface{}, minVersion string) {
+	id := g.mu.RLock()
+	defer g.mu.RUnlock(id)
```

Comment:
It's very, very important that module locks not be held during network calls. That's a severe DoS vector, and in general will have a massive impact on performance and usability.